### PR TITLE
rucola: init at 0.4.1

### PR DESCRIPTION
**Pull request by donovanglover**
## Description of changes

Adds [rucola](https://github.com/Linus-Mussmaecher/rucola), a terminal-based markdown note manager written in Rust.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by devhell** [comment]
Result of `nixpkgs-review pr 341286` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>rucola</li>
  </ul>
</details>

Can this be merged?

**Comment by donovanglover** [comment]
I set `platforms = lib.platforms.linux;` since it was failing on Darwin. Contributions welcome if anyone knows how to add Darwin support.

**Comment by devhell** [comment]
I have no way of testing for Darwin, but until someone can, perhaps this can be merged now?

**Comment by afh** [comment]
I'll have a look @devhell

File: pkgs/by-name/ru/rucola/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  };
+
+  meta = {
+    description = "Terminal-based markdown note manager";
+    homepage = "https://github.com/Linus-Mussmaecher/rucola";
+    changelog = "https://github.com/Linus-Mussmaecher/rucola/blob/${src.rev}/CHANGELOG.md";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ donovanglover ];
+    mainProgram = "rucola";
+    platforms = lib.platforms.linux;
```
Comment (afh) [inline pkgs/by-name/ru/rucola/package.nix:41]:
```suggestion
    platforms = lib.platforms.linux ++ lib.platforms.darwin;
```

File: pkgs/by-name/ru/rucola/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  cargoHash = "sha256-KP28bDTNv4NmkQyE0g3GGsgzJt6MPx+ZRH6YJbiv0H4=";
+
+  nativeBuildInputs = [
+    pkg-config
+  ];
+
+  buildInputs = [
+    oniguruma
+  ];
```
Comment (afh) [inline pkgs/by-name/ru/rucola/package.nix:28]:
```suggestion
  ] ++ lib.optionals stdenv.hostPlatform.isDarwin (with darwin.apple_sdk.frameworks; [
    CoreServices SystemConfiguration
  ]);
```

File: pkgs/by-name/ru/rucola/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  pkg-config,
+  oniguruma,
```
Comment (afh) [inline pkgs/by-name/ru/rucola/package.nix:6]:
```suggestion
  oniguruma,
  darwin,
  stdenv,
```

**Review by afh — CHANGES_REQUESTED** [review]
The following suggestions below are necessary in order to build rucola on darwin.

Not sure why but the 
`test io::file_tracker::tests::test_watcher_rename` test fails.
Any ideas?

**Comment by donovanglover** [comment]
No idea but upstream doesn't seem to test Darwin in CI so I'm fine with skipping it

**Comment by Aleksanaa** [comment]
> No idea but upstream doesn't seem to test Darwin in CI so I'm fine with skipping it

Could you mark it as broken?

**Review by Aleksanaa — APPROVED** [review]
Others seems fine

**Comment by afh** [comment]
@donovanglover is there a way to disable the faulty test in the rucola nixpkg when building on darwin? I'd prefer that over marking it as broken.

Any objections, @Aleksanaa?

**Comment by Aleksanaa** [comment]
I'll take a look

**Comment by devhell** [comment]
What's the status on this PR?

File: pkgs/by-name/ru/rucola/package.nix [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  buildInputs =
+    [
+      oniguruma
+    ]
+    ++ lib.optionals stdenv.hostPlatform.isDarwin (
+      with darwin.apple_sdk.frameworks;
+      [
+        CoreServices
+        SystemConfiguration
+      ]
```
Comment (Aleksanaa) [inline pkgs/by-name/ru/rucola/package.nix:37]:
```suggestion
      [
        apple_sdk_11
        (darwinMinVersionHook "10.13")
      ]
```

Comment (donovanglover) [reply pkgs/by-name/ru/rucola/package.nix]: Used `apple-sdk_11`

**Comment by donovanglover** [comment]
Looks like the build passes on darwin now :tada:

**Review by Aleksanaa — APPROVED** [review]

**Pull request merged**